File: SDKs/sportsdata.io/mlb-v3-stats/1.0/python/src/sdk/models/operations/playerseasonstatssplitbyteam.py
from dataclasses import dataclass, field
from typing import Any,List,Optional
from enum import Enum

class PlayerSeasonStatsSplitByTeamFormatEnum(str, Enum):
    XML = "XML"
    JSON = "JSON"


@dataclass
class PlayerSeasonStatsSplitByTeamPathParams:
    format: PlayerSeasonStatsSplitByTeamFormatEnum = field(metadata={'path_param': { 'field_name': 'format', 'style': 'simple', 'explode': False }})
    season: str = field(metadata={'path_param': { 'field_name': 'season', 'style': 'simple', 'explode': False }})
    

@dataclass
class PlayerSeasonStatsSplitByTeamRequest:
    path_params: PlayerSeasonStatsSplitByTeamPathParams = field()
    

@dataclass
class PlayerSeasonStatsSplitByTeamResponse:
    content_type: str = field()
    status_code: int = field()
    player_seasons: Optional[List[Any]] = field(default=None)
    

File: SDKs/sportsdata.io/mlb-v3-stats/1.0/typescript/src/sdk/models/operations/playerseasonstatssplitbyteam.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum PlayerSeasonStatsSplitByTeamFormatEnum {
    Xml = "XML",
    Json = "JSON"
}


export class PlayerSeasonStatsSplitByTeamPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=format" })
  format: PlayerSeasonStatsSplitByTeamFormatEnum;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=season" })
  season: string;
}


export class PlayerSeasonStatsSplitByTeamRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: PlayerSeasonStatsSplitByTeamPathParams;
}


export class PlayerSeasonStatsSplitByTeamResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  playerSeasons?: any[];

  @SpeakeasyMetadata()
  statusCode: number;
}
